app/components/DataTable/components/ActionCellRenderer/__tests__/ActionCellRenderer.test.tsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import ActionCellRenderer from '..';
+
+// eslint-disable-next-line react/display-name
+jest.mock('../../SvgViewer', () => (() => (<div role="presentation" />)));
+// eslint-disable-next-line react/display-name
+jest.mock('../../ImageViewer', () => (() => (<div role="presentation" />)));
+
+const mockSvg = '<svg><g>this is an svg</g></svg>';
+const mockImage = 'data:PNG;base64, image data here';
+
+describe('ActionCellRenderer', () => {
+  test('It matches the snapshot', () => {
+    const { asFragment } = render(
+      <ActionCellRenderer data={{}} />,
+    );
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('It shows the detail icon when allowed', async () => {
+    render(
+      <ActionCellRenderer
```

Comment:
a lot of these types are yelling pretty hard, is it worth it to use this with an `AgGrid`?  it'll keep yelling until the typeDefs are updated to the correct type